stix_shifter/stix_translation/src/modules/guardium/query_constructor.py

```diff
@@ -0,0 +1,524 @@
+                # Insert the param in the curResObj
+                print(param)
+                if param not in cp_curResObj:
+                    cp_curResObj[param] = thisJObj[param]
+                    #print(cp_curResObj)
+                    if (curPos + 1) < pArrSize:
+                        resArray = self.buildArrayOfGuardiumReportParams(resArray, resPos, cp_curResObj, paramsArray, curPos)
+                    else:
+                        resArray.append(cp_curResObj)
+                        resPos = resPos + 1
+        #
+        #print(resPos)
+        #print(resArray)
+        return resArray
+    #
+    def substitute_ParamsPassed(self, reportDefs, reports_in_query):
+    # for Each report in reportDefs substitue params for report Params Passed
+    #   generate all reports for the query
+        logging.info("-------- Substitute Params Passed ------\n reportDefs")
+        logging.info(reportDefs)
+        logging.info("---- REPORT PROCESSING START ---\n")
+        for reportName in reportDefs:
+            report = reportDefs[reportName]
+            logging.info("processing ... " + str(reportName) + "\n")
+            logging.debug(report)
+            print(report)
+            for param in report["reportParameter"]:
+                print(param)
+                # either the value will be default or passed by ISC (report parameter passed)
+                if param not in self.reportParamsPassed:
+                    value = report["reportParameter"][param]["default"]
+                else:
+                    value = self.reportParamsPassed[param]
+                print(value)
+                print(report["reportParameter"][param]["info"])
+     #
+     # Use START and STOP  instead of default to time parameter
+                if report["reportParameter"][param]["info"] == "START":
+                    value = self.reportParamsPassed["START"]
+                    print("START")
+                    print(value)
+                if report["reportParameter"][param]["info"] == "STOP":
+                    value = self.reportParamsPassed["STOP"]
+                    print("STOP")
+                    print(value)
+      #
+      #Transform the value or use it as-is
+                if "transformer" in report["reportParameter"][param]:
+                    transformer = self.transformers[report["reportParameter"][param]["transformer"]]
+                    report["reportParameter"][param] = transformer.transform(value)
+                else:
+                    report["reportParameter"][param] = value
+
+            logging.info("processing ... Completed: " + str(reportName) + "\n")
+            logging.debug(report)
+            reports_in_query.append(report)
+        logging.info("-------- End Substitute Params Passed ------\n")
+#
+        return reports_in_query
+#
+    def get_report_params(self):
+        reports_in_query = []
+        for repParamIndex in range(self.reportParamsArraySize):
```

Comment:
Why not just do `for report in reportParamsArray:`?  You don't need to iterate by index. Also, you keep overwriting the `self.reportParamsPassed` attribute on each iteration. Since this is initialized as an empty object, it seems like something that should be updated rather than overwritten.